packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -96,6 +97,68 @@ class _ScaffoldLayout extends MultiChildLayoutDelegate {
   bool shouldRelayout(MultiChildLayoutDelegate oldDelegate) => false;
 }
 
+class _FloatingActionButtonTransition extends StatefulComponent {
+  _FloatingActionButtonTransition({
+    Key key,
+    this.child
+  }) : super(key: key) {
+    assert(child != null);
+  }
+
+  final Widget child;
+
+  _FloatingActionButtonTransitionState createState() => new _FloatingActionButtonTransitionState();
+}
+
+class _FloatingActionButtonTransitionState extends State<_FloatingActionButtonTransition> {
+  final Performance hidePerformance = new Performance(duration: _kFloatingActionButtonSegue);
+  final Performance showPerformance = new Performance(duration: _kFloatingActionButtonSegue);
+  Widget oldChild;
+
+  void initState() {
+    super.initState();
+    showPerformance.play();
+  }
+
+  void dispose() {
+    hidePerformance.stop();
+    showPerformance.stop();
+    super.dispose();
+  }
+
+  void didUpdateConfig(_FloatingActionButtonTransition oldConfig) {
+    if (config.child.key == oldConfig.child.key)
+      return;
+    oldChild = oldConfig.child;
+    showPerformance.progress = 0.0;
+    hidePerformance
+      ..progress = 0.0
+      ..play().then((_) {
+        oldChild = null;
+        showPerformance.play();
+     });
+  }
+
+  Widget build(BuildContext context) {
+    List<Widget> children = new List<Widget>();
+
+    if (oldChild != null)
+      children.add(new ScaleTransition(
+        scale: new AnimatedValue<double>(1.0, end: 0.0, curve: Curves.easeIn),
+        performance: hidePerformance,
+        child: oldChild
+      ));
+
+    children.add(new ScaleTransition(
+      scale: new AnimatedValue<double>(0.0, end: 1.0, curve: Curves.easeIn),
+      performance: showPerformance,
+      child: config.child
+    ));
+
+    return new Stack(children);
```

Comment:
What's the point of stacking them?  Can they ever both be visible at the same time?  Seems like you want to build one or the other.